spec/datadog/tracing/tracer_spec.rb

```diff
@@ -598,6 +618,23 @@
           expect(child.end_time).to be > parent.end_time
         end
       end
+
+      context 'for span sampling' do
+        let(:tracer_options) { super().merge(span_sampler: span_sampler) }
+        let(:span_sampler) { instance_double(Datadog::Tracing::Sampling::Span::Sampler) }
+
+        before do
+          allow(span_sampler).to receive(:sample!)
+        end
+
+        it 'invokes the span sampler with the current span and trace operation' do
+          span_op = trace
+          trace_op = tracer.active_trace
+          span = span_op.finish
+
+          expect(span_sampler).to have_received(:sample!).with(trace_op, span)
```

Comment:
Minor: I'm curious why in the other test in line 555 we use `@span_op.send(:build_span)` but in here we do `span_op.finish`?

Reply:
They effectively return the same value, which is the result of `#build_span`, but `#finish` has more side effects.

I think the correct thing is to call finish in both in this PR, I'll do that. :thankyou: